### PR TITLE
Fix render_mode when loading VecNormalize

### PR DESCRIPTION
**Pull request by araffin**
## Description

See https://github.com/DLR-RM/rl-baselines3-zoo/issues/399

I've also switched to ruff in place of isort (everything is faster now, for the same result and less dependencies =)).

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) for new features and bug fixes)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)

## Checklist

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [ ] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.
- [ ] I have opened an associated PR on the [SB3-Contrib repository](https://github.com/Stable-Baselines-Team/stable-baselines3-contrib) (if necessary)
- [ ] I have opened an associated PR on the [RL-Zoo3 repository](https://github.com/DLR-RM/rl-baselines3-zoo) (if necessary)
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` and `make type` both pass. (**required**)
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -120,12 +120,10 @@
...
             "mypy",
-            # Lint code (flake8 replacement)
-            "ruff",
-            # Sort imports
-            "isort>=5.0",
+            # Lint code and sort imports (flake8 and isort replacement)
+            "ruff>=0.0.288",
             # Reformat
-            "black",
+            "black>=23.9.1,<24",
```
Comment (ernestum) [inline setup.py:126]: Btw `black~=23.9` should do the same here

Comment (araffin) [reply setup.py:126]: yes it is, I do find `>=23.9.1,<24` more explicit though (the thing is `~=` is not consistent among package managers/languages... :/ see https://jubianchi.github.io/semver-check/#/~23.9/23.10 for instance).

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -38,7 +38,7 @@ pip install -e .[docs,tests,extra]
 
 ## Codestyle
 
-We use [black codestyle](https://github.com/psf/black) (max line length of 127 characters) together with [isort](https://github.com/timothycrosley/isort) to sort the imports.
+We use [black codestyle](https://github.com/psf/black) (max line length of 127 characters) together with [ruff](https://github.com/astral-sh/ruff) (isort rules) to sort the imports.
```
[the remark this answers is not in the document]
Comment (araffin) [reply CONTRIBUTING.md:41]: ruff is already used as a replacement for flake8 and it's pretty fast =)

File: setup.py (hunk 1 above)
Comment (ernestum) [reply setup.py:126]: Oh I did not know this! Is this also the case in the python world?

**Pull request merged**